ieeeuoft/hackathon-template
# RSVP View Backend Implementation

### Overview
- Resolves #180 
- Implements a view that takes in either a `rsvp=yes` or `rsvp=no` `HTTP GET` parameters
- In the case that the request goes through, the user either
   - Gets `application.rsvp=True` and a profile created, if they `rsvp=yes`
   - Gets `application.rsvp=False`, if they `rsvp=no`
- The user cannot RSVP if the date is later than `RSVP_DAYS` past the date that a decision was sent to them. Additionally, at any time in the `RSVP_DAYS` time period, the user can change their mind and RSVP no
- The user cannot RSVP if they:
   1. Don't have an application
   2. Haven't had their application reviewed
   3. Haven't been accepted

   These result in `400` responses.
- Note that if we put their application status on the dashboard, we can just redirect to the dashboard for case `3`
- Otherwise we redirect to the dashboard immediately.

### Testing
- Unit tests for `400` responses, yes RSVP, no RSVP, and RSVP deadline passed

### QA
1. Create a user and application, and a review for yourself in the admin site. Then login to that user.
2. Go to `/registration/rsvp/yes` to RSVP yes, `/registration/rsvp/no` to RSVP no
   1. After you RSVP yes, go to admin confirm that you have a profile
   2. After you RSVP yes, go to admin confirm that you DO NOT have a profile
3. Try giving yourself a different `status` other than `Accepted` and see what happens
4. Try going to `/registration/rsvp/X` where `X=anything other than yes or no` - this should `404`
